content/articles/Servlet-Filter-API/index.md

```diff
@@ -0,0 +1,330 @@
+1. Create a Java class that implements the interface `Filter`.
+2. Set up the filter parameters.
+3. Finally, do a filter mapping.
+
+We must implement the Servlet 'Filter' Interface, which is part of the `javax.servlet`  package, to create a filter, and the life cycle methods for the 'Filter' interface are as follows:
+
+ - `void init()`: The 'init()' method is used to initialize Filter parameters; it informs the web container that a filter needs to be activated. 
+Only one parameter is required, namely,
+
+```java
+public void init(FilterConfig parameterName) throws ServletException
+
+```
+ - `void doFilters()`:  This method is the most significant in the Filter interface and calls the next filter. It requires three parameters: 
+- ServletRequest obj
+- ServletResponse obj
+- FilterChain obj, implemented as follows.
+```java
+ Public void doFilters(ServletRequest request, ServletResponse response,FileterChain chain)throws ServletException,IOException
+```
+
+- `void destroy()`: This method denotes that the filter has completed its duty or has been removed from service. Users can override this method to write finishing logic, such as releasing resources, objects, and so on.
+
+### FilterChain
+FilterChain's object is passed to the Filter interfaces when executing the next filter or resource in the chain.
+#### Method.
+
+- `void doFilter()`: This method is only available in the `FilterChain` interface and used when calling the next `Filter` and passing all resources that are required:
+```java
+public void doFilter(HttpServletRequest request, HttpServletResponse response);
+```
+### FilterConfig
+
+The web container generates a `FilterConfig` object which is used to retrieve configuration information from the web.xml file.
+
+#### FilterConfig interface methods
+In the FilterConfig interface, there are four methods:
+1. `void init()`: This method is only called once and is used to set up the filter.
+2. `String getInitParameter`: The parameter value for the parameter name provided is returned by this method.
+3. `getInitParameterNames()`: This method returns a list of all type enumeration parameter names.
+4. `getServletContext()`: This method returns the `ServletContext` object.
+
+### Authentication Example using filter
+In this example, we're demonstrating how a `Filter` is used to achieve authentication. We're using two filters here: one to verify if a password is greater than seven characters, and the other to check both the user name and password supplied by the user.
+Filter 1 will forward the request to a second Filter if the password length is correct. Otherwise, an error notice will appear.
+
+If the user name and password provided by the user are both correct, the second Filter will route the request to the Servlet page; otherwise, an error message will be displayed.
+#### Steps to create Authentication Example
+We assumed you already knew how to use the eclipse IDE and configure the Tomcat server in this example.
+
+#### Step 1:

```

Comment:
Please give a brief title for this steps and all the other steps i.e
Step one: Creating a file